pkg/tagger/collectors/kubelet_extract.go

```diff
@@ -106,10 +100,33 @@ func (c *KubeletCollector) parsePods(pods []*kubelet.Pod) ([]*TagInfo, error) {
 // parseDeploymentForReplicaset gets the deployment name from a replicaset,
 // or returns an empty string if no parent deployment is found.
 func (c *KubeletCollector) parseDeploymentForReplicaset(name string) string {
-	// TODO optionaly query the cluster agent instead of assuming?
-	match := KubeReplicaSetRegexp.FindStringSubmatch(name)
-	if len(match) < 2 {
+	lastDash := strings.LastIndexAny(name, "-")
+	if lastDash == -1 {
+		// No dash
+		return ""
+	}
+	suffix := name[lastDash+1:]
+	if len(suffix) < 3 {
+		// Suffix is variable length but we cutoff at 3+ characters
+		return ""
+	}
+
+	if !stringinRuneset(suffix, Digits) && !stringinRuneset(suffix, KubeAllowedEncodeStringAlphaNums) {
```

Comment:
Can't we use `HasSuffix` directly?

Reply:
I don't think, because we want the whole `suffix` slice (delimited by the dash we found before) to match either one of these runesets.
If we want to avoid creating `stringinRuneset`, we could `TrimSuffix` then make sure len == 0, but I don't think that's very explicit